experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -158,6 +174,18 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if !inputcr.GetDeletionTimestamp().IsZero() {
+		return r.reconcileDelete(ctx, logger, inputcr)
+	}
+	// Add a finalizer to prevent removal of facade before all applied objects are cleaned up.
+	if !controllerutil.ContainsFinalizer(&inputcr, finalizerName) {
+		controllerutil.AddFinalizer(&inputcr, finalizerName)
```

Comment:
Do we need a finalizer for the plan object ? If we add a finalizer here ?

Reply:
Finalizers on both resources won't prevent one another from being deleted. The Plan finalizer is mostly to keep the plan from being accidentally removed out of band, which would cause big problems for facade deletion.